### PR TITLE
SY-4033: Add Copy Diagnostics to Status Notifications

### PR DESCRIPTION
**Pull request by emilbon99**
# Issue Pull Request

## Linear Issue

[SY-4033](https://linear.app/synnaxlabs/issue/SY-4033)

## Description

Adds standardized clipboard copy infrastructure across the console and Pluto:

- **`Button.Copy`** now auto-pushes status notifications on success (via `successMessage` prop) and errors (via `handleError`). Supports async text functions. When a child is a React element (e.g. an icon), it replaces the default copy icon and swaps to a checkmark on success. When `successMessage` is provided, the status notification replaces the checkmark feedback.
- **`Menu.CopyItem`** - new Pluto component wrapping `Button.Copy` styled as a menu item for use in context menus.
- **Status notification toast** - added a copy diagnostics button that appears on hover.
- **Task controls status** - existing copy diagnostics button now uses the shared infrastructure.
- **Console consumers migrated** - task utility buttons, task details header, range overview, line plot context menu, ontology copy properties, and status explorer context menu all now use `Button.Copy` or `Menu.CopyItem` instead of manual `navigator.clipboard` calls.
- **Consistent divider grouping** across task and range pages: copy-as-code | copy-data | share.
- **Task utility buttons** now conditionally render based on whether the task has been saved (no key = only JSON config copy shown).

## Basic Readiness

- [x] I have performed a self-review of my code.
- [x] I have added relevant, automated tests to cover the changes.
- [ ] I have updated documentation to reflect the changes.

<h3>Greptile Summary</h3>

This PR introduces reusable `Button.Copy` and `Menu.CopyItem` components backed by the status aggregator for error handling and success feedback, and systematically migrates manual `navigator.clipboard` calls across the console. The implementation is well-structured and the test coverage is thorough. Three P2 style/improvement issues were found: `onCopy` is silently dropped when `successMessage` is set (an early `return` exits before the callback fires), the `{...rest}` spread in `CopyItem` can silently override the menu context's click handler, and a test description contains a minor typo. None of these block merging.

<h3>Confidence Score: 5/5</h3>

This PR is safe to merge — all findings are minor style/API consistency improvements.

All three identified issues are P2: one is a subtle callback-omission edge case with no current consumer affected, one is an API footgun with no current exploiter, and one is a test description typo. No data integrity, security, or correctness issues were found.

pluto/src/button/Copy.tsx (onCopy/successMessage interaction), pluto/src/menu/CopyItem.tsx (onClick spread order)

<details open><summary><h3>Vulnerabilities</h3></summary>

No security concerns identified. Clipboard writes use the standard `navigator.clipboard.writeText` API (browser-sandboxed). Errors are handled by the status aggregator without exposing raw stack traces in the UI. No user-supplied data is executed as code.
</details>

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| pluto/src/button/Copy.tsx | New Button.Copy component with clipboard write, success/error status integration via Aggregator, and checkmark feedback; onCopy/successMessage interaction is a minor inconsistency |
| pluto/src/button/Copy.spec.tsx | Comprehensive tests covering rendering, clipboard copying (sync/async/string), callbacks, status notification push, and button prop passthrough |
| pluto/src/menu/CopyItem.tsx | New Menu.CopyItem wrapping Button.Copy as a styled menu item; onClick spread order creates a minor API footgun with no current impact |
| pluto/src/menu/CopyItem.spec.tsx | Tests for rendering children, clipboard copying, async text functions, menu context onChange, and CSS class application |
| pluto/src/menu/external.ts | Adds CopyItem to menu package exports |
| pluto/src/status/base/Notification.tsx | Adds hover-visible copy diagnostics button (Button.Copy) to status toast header, serializing the notification via status.toString |
| pluto/src/status/base/Notification.css | Hides header-actions by default and reveals them on notification hover for a clean progressive-disclosure UX |
| pluto/src/text/Text.tsx | Minor refactor extracting formatStyle helper for lineClamp CSS application |
| x/ts/src/status/status.ts | Adds toString (with JSON pretty-print, stack trace, and extra details formatting), fromException, keepVariants, and removeVariants utilities |
| x/ts/src/status/status.spec.ts | Thorough tests for all new utilities across all variant types and option combinations; one typo in a test description (undefindd) |
| console/src/hardware/common/task/UtilityButtons.tsx | Migrates manual clipboard calls to Button.Copy with successMessage; code-copy buttons now conditionally render based on whether the task has been saved |
| console/src/hardware/common/task/layouts/DetailsHeader.tsx | Migrates copy details button to Button.Copy with successMessage, removing manual navigator.clipboard call |
| console/src/lineplot/LinePlot.tsx | Migrates context menu copy-range items to Menu.CopyItem with successMessage notifications |
| console/src/ontology/CopyPropertiesContextMenuItem.tsx | Simplified to Menu.CopyItem, removing manual clipboard call and navigator usage |
| console/src/range/overview/Details.tsx | Migrates Python/TypeScript code copy buttons to Button.Copy with per-name success messages |
| console/src/range/overview/MetaData.tsx | Migrates inline copy button in metadata value input to Button.Copy with successMessage |
| console/src/status/list/ContextMenu.tsx | Migrates copy diagnostics context menu item to Menu.CopyItem with successMessage |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant User
    participant ButtonCopy as Button.Copy
    participant Clipboard as navigator.clipboard
    participant Aggregator as Status.Aggregator
    participant UI as UI State

    User->>ButtonCopy: click
    ButtonCopy->>ButtonCopy: resolve text (sync/async fn or string)
    ButtonCopy->>Clipboard: writeText(resolvedText)
    alt successMessage provided
        Clipboard-->>ButtonCopy: success
        ButtonCopy->>Aggregator: addStatus({ variant: success, message })
        Aggregator-->>UI: toast notification
    else no successMessage
        Clipboard-->>ButtonCopy: success
        ButtonCopy->>UI: setCopied(true) — show checkmark
        ButtonCopy->>ButtonCopy: setTimeout — setCopied(false)
    end
    alt clipboard write fails
        Clipboard-->>ButtonCopy: Error
        ButtonCopy->>Aggregator: addStatus({ variant: error, message: err.message })
        Aggregator-->>UI: error toast notification
    end
```

<sub>Reviews (1): Last reviewed commit: ["checkpoint"](https://github.com/synnaxlabs/synnax/commit/9b1cfa5e068820993a393e63b216d287d8ecc3f2) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27657548)</sub>

> Greptile also left **3 inline comments** on this PR.

### CODE REVIEW

File: pluto/src/status/base/Notification.tsx [hunk 1]
```diff
@@ -84,14 +86,30 @@ export const Notification = ({
...
-          size="small"
-          onClick={() => silence(key)}
+        <Flex.Box
+          x
+          className={CSS(CSS.BE("notification", "header-actions"))}
+          gap="tiny"
         >
-          <Icon.Close />
-        </Button.Button>
+          <Button.Copy
```
Comment (pjdotson) [inline pluto/src/status/base/Notification.tsx:94]: Would be nice to have a test for this

**Review by pjdotson — CHANGES_REQUESTED** [review]
Approved after comment addressed
